server/sonar-server/src/main/java/org/sonar/server/computation/filemove/MatchesByScore.java

```diff
@@ -21,56 +21,81 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
 
 import static org.sonar.server.computation.filemove.FileMoveDetectionStep.MIN_REQUIRED_SCORE;
 
-class MatchesByScore implements ScoreMatrix.ScoreMatrixVisitor, Iterable<List<Match>> {
-  private final ScoreMatrix scoreMatrix;
-  private final List<Match>[] matches;
-  private int totalMatches = 0;
-
-  private MatchesByScore(ScoreMatrix scoreMatrix) {
-    this.scoreMatrix = scoreMatrix;
-    this.matches = new List[Math.max(MIN_REQUIRED_SCORE, scoreMatrix.getMaxScore()) - MIN_REQUIRED_SCORE];
-  }
+abstract class MatchesByScore implements Iterable<List<Match>> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Convert the abstract class "MatchesByScore" into an interface. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1610)